fastapi/param_functions.py

```diff
@@ -239,15 +241,19 @@ def File(  # noqa: N802
 
 
 def Depends(  # noqa: N802
-    dependency: Optional[Callable[..., Any]] = None, *, use_cache: bool = True
-) -> Any:
-    return params.Depends(dependency=dependency, use_cache=use_cache)
+    dependency: Optional[Callable[..., _T]] = None, *, use_cache: bool = True
+) -> _T:
+    return params.Depends(dependency=dependency, use_cache=use_cache)  # type: ignore
```

Comment:
I'm curious. Why not propagate the _T to the Depends? I mean, define a T (or another name there and import on this file as well) 

I'm asking this because of the ignore comment.

Reply:
I'm no expert. `Depends` is a class:
https://github.com/tiangolo/fastapi/blob/33be5fc8baa02bae949e1986f1821dae8f84f487/fastapi/params.py#L316
How do I type-override a class?

Reply:
I think you can just make the class generic:
```py
from typing import TypeVar, Generic

T = TypeVar("T")


class Depends(Generic[T]):
    def __init__(
        self, dependency: Optional[Callable[..., T]] = None, *, use_cache: bool = True
    ):
        self.dependency = dependency
        self.use_cache = use_cache
    ...
```